### PR TITLE
doc: update documentation for macOs

### PR DESCRIPTION
**Pull request by ghost**
##### Checklist

- [x ] documentation is changed or added

##### Affected core subsystem(s)

None

##### Description of change

Add documentation for installing node-chakracore on Mac OS

### CODE REVIEW

**Comment by kunalspathak** [comment]
Thanks @jackTheRipper for the contribution. @jianchun , @obastemur - can you review this?

File: README.md [hunk 1]
```diff
@@ -116,6 +116,39 @@ $ node -e "console.log('Hello from Node.js ' + process.jsEngine)"
 Hello from Node.js chakracore
 ```
 
+#### Mac OS
+
+Prerequisites:
+* XCode v8.2.1
```
Comment (obastemur) [inline README.md]: XCode 7+ should be fine.

File: README.md [hunk 2]
```diff
@@ -116,6 +116,39 @@ $ node -e "console.log('Hello from Node.js ' + process.jsEngine)"
...
+Clone this repo and check out **`xplat`** branch:
+
+```
+git checkout xplat
+```
+
+Build:
+
+```
+./configure --engine=chakracore --dest-os=mac --with-icu-path=/usr/local/opt/icu4c/include --without-intl
```
Comment (obastemur) [inline README.md]: is there a reason for `--dest-os=mac` here?

**Review by obastemur — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ghost) [reply README.md]: I haven't tried myself. If you think it would be fine I will update it

File: README.md (hunk 2 above)
Comment (ghost) [reply README.md]: yes, I was experimenting. I will try to re build the Chakra core without the flag.

Comment (ghost) [reply README.md]: It works without the flag.

File: README.md (hunk 1 above)
Comment (ghost) [reply README.md]: updated

**Review by obastemur — APPROVED** [review]

**Comment by obastemur** [comment]
@jackTheRipper Thanks, could you please squash?

**Comment by ghost** [comment]
Sure. Done

**Comment by jasonkarns** [comment]
Are the `--with-icu-path=/usr/local/opt/icu4c/include --without-intl` configure options still supposed to be necessary on OSX? I just did a build (on 10.12.3) without them (matching the Linux build steps) and it compiled successfully (and process.jsEngine is correct).

**Comment by ghost** [comment]
@jasonkarns I performed a test recently in order to use the Node-Chakracore with NVM and I had to add this flags.

How did you manage with the `intl` ? Did you setup your environment for `icu` ?

**Comment by jasonkarns** [comment]
Honestly, all I did was `configure` (no options) and `make` then asserted the build with ` ./node -e "console.log('Hello from Node.js ' + process.jsEngine)"`. Are the flags necessary to even compile? Or might I get a runtime error at some point from this build?

**Comment by obastemur** [comment]
@jasonkarns I'm not sure how your compilation process went well but i.e icu option is needed because of https://github.com/nodejs/node-chakracore/blob/xplat/deps/chakrashim/chakracore.gyp#L86

**Comment by jasonkarns** [comment]
It defaults to empty string. What indicates an actual compilation failure would occur when that value is empty?

**Comment by obastemur** [comment]
It would use `icucore` from the system. However that may end up giving a linker error. We could actually default to `--no-icu` when `--icu` wasn't set. @digitalinfinity @kunalspathak  ?

**Comment by jasonkarns** [comment]
If defaulting to `icucore` _may_ give a linker error, what are the chances that it doesn't at compile time? Is there a known module or package I could run in a chakracore-compiled node that would force usage of icu? Such that the node compiled without the icu-path would then give a runtime error or crash?

**Comment by obastemur** [comment]
if it doesn't give anything on compile time, that means linking was successful. Otherwise you wouldn't have the binary.

**Pull request merged**